### PR TITLE
Dark mode support

### PR DESCRIPTION
**Pull request by nCastle1**
Adds dark mode support for Android, iOS, and UWP. Also makes other minor changes to support modern features like dark mode.

### CODE REVIEW

File: src/MapsApp.Xamarin.Shared/Views/StartPage.xaml [hunk 1]
```diff
@@ -309,24 +292,24 @@
...
             <Grid.RowDefinitions>
                 <RowDefinition Height="Auto" />
                 <RowDefinition Height="*" />
             </Grid.RowDefinitions>
             <Button Text="Go back" Grid.Row="0" 
-                    BackgroundColor="White" TextColor="Accent" 
+                    TextColor="{DynamicResource AccentColor}" 
                     Clicked="NavigatedView_CloseClicked" />
             <!-- Loading indicator to show when other lists aren't ready yet -->
             <ActivityIndicator Grid.Row="1" IsRunning="True" />
```
Comment (ZackAllen) [inline src/MapsApp.Xamarin.Shared/Views/StartPage.xaml:315]: This indicator displays as too large on Android, filling the whole screen. Setting a width/height for android might fix this.

**Review by ZackAllen — APPROVED** [review]
Overall looks good. No major issues.

**Comment by nCastle1** [comment]
There is a limitation where listview don't update properly when the system theme changes. In practice that only seems to affect the directions page. There is a Xamarin.Forms bug logged and it looks like they'll be addressing it soon, so I suggest moving forward with this PR and waiting for the Forms bug to be fixed.

https://github.com/xamarin/Xamarin.Forms/issues/11750

**Review by esreli — COMMENTED** [review]
This is no small feat, thanks for your help here. I'll need to divide up this review into multiple, one per platform. Also, being that these changes are design-focused, this review will offer UI/design adjacent feedback in addition to reviewing dark/light mode.

## iOS

I was unable to fully review the app because I encountered this [bug](https://github.com/ArcGIS/open-source-apps/issues/446).

### Light/Dark Mode

Do the colors match the colors stated in the [style guide](https://github.com/ArcGIS/open-source-apps/wiki/Platform-Styleguides#app-narrative-colors)? It appears they are the same hue of red in either light and dark modes. Would you update the colors with those specified in the style guide and swap them dependent on whether the device is in light or dark mode? In iOS this can be accomplished fairly easily using an xcasset catalog color.

### Generate Route Glyph

Would you please modify the generate route glyph such that the glyph is the app's theme color (red) rather then what it is now where the background is the app's theme color and the glyph is black?

Screen Shot 2020-09-08 at 2 53 38 PM
Screen Shot 2020-09-08 at 2 51 20 PM

### Notification Action Button

A notification button uses the app's theme color as the background. Can you please remove the background color and instead change the button's text color to the app's theme color (red).

Screen Shot 2020-09-08 at 2 55 04 PM

### Safe Area

It doesn't appear there is support here for devices with safe areas. Can you introduce this support?

Screen Shot 2020-09-08 at 3 28 57 PM

**Review by esreli — COMMENTED** [review]
Thanks @nCastle1, I think we're nearly there. I've captured a few suggestions, divided into two categories. The first, **UI Changes**, contains some suggestions and some questions. Feel free to dismiss anything you feel is invalid or cumbersome. The second, **Bugs**, are bugs i've encountered testing the app in the Android Simulator. I'm submitting them here because I discovered them while reviewing this work, feel free to break them out into individual issues if you feel as though they merit separate attention.

Please note, I chose to test using an Android simulator because of the stated added challenges in supporting the various quasi-safe areas specified in the Android SDK.

_I performed this review using the Android Simulator, simulating the Pixel 2 XL, using API 29_

---

## UI Changes

### Matching 'Safe Area' Background Colors

I think the solution you've come up with regarding how to handle the safe area colors is a good one. I'm probably missing something here but i've got just one question here, is there a reason the background colors don't match? As it is currently built, the white background does not offer much contrast to the Android device system colors. Would you consider change the bottom safe area background color to match the top?

Screen Shot 2020-09-14 at 1 08 13 PM

### Table Item Label Color

According to the Apple HIG [Color section](https://developer.apple.com/design/human-interface-guidelines/ios/visual-design/color/),

> **Use color judiciously for communication.** The power of color to call attention to important information is heightened when used sparingly. For example, a red triangle that warns people of a critical problem becomes less effective when red is used elsewhere in an app for noncritical reasons.

I understand this design decision affects the Android build only but I think the sentiment remains true. A theme color should be used judiciously, I fear using the theme color to color list labels is noncritical and the theme color's value is then lost. Would you please consider changing the label color to a grayscale color, contrasting to the system light/dark mode preference? The same should be done for the route step icon.

| Search Results | Directions List |
| --- | --- |
| Screen Shot 2020-09-14 at 1 11 19 PM | Screen Shot 2020-09-14 at 1 24 52 PM |


### (Optional) Hamburger Button Left Padding

_Please address this item only if it is a simple change (i.e. 10< minutes)._

Can we adjust the left padding of the Hamburger Button Icon? I feel like the icon's padding could be almost cut in half. Ideally, the padding would match that of the 'X' (close) button located on the right side of the search bar.

Screen Shot 2020-09-14 at 1 09 57 PM


### (Optional) Directions Button Right Aligned

_Please address this item only if it is a simple change (i.e. 10< minutes)._

Can we adjust the positioning of the two buttons contained in the bottom result view? It appears there is a substantial amount of padding-right between the 'X' (close) button and the edge of the view, there's even more padding-right between the directions button and the 'X' (close) button. Is this an intentional style decision? It feels as though this style cramps the bottom bar, scrunching the result address. Can we apply a more standard or default padding here, keeping these two buttons as right-aligned as possible?

Screen Shot 2020-09-14 at 1 19 04 PM

## Bugs 

### Attribution Bar Unresponsive

It appears the Attribution Bar is unresponsive to my (simulated) tap. In the case where the attribution text is suffixed by an ellipsis I am making the assumption there is more text that can be revealed though tapping the attribution bar has no effect.

### Basemap Picker Thumbnail

I've encountered a bug where it doesn't appear to load the thumbnail associated with a basemap. Is the thumbnail supposed to load?

Screen Shot 2020-09-14 at 3 44 59 PM

Also **optionally**, can this view be popped onto the view stack and presented in the same way viewing the list of directions is? _Please address this item only if it is a simple change (i.e. 10< minutes)._

**Review by esreli — APPROVED** [review]
You've captured all of my feedback nicely! I see no further issues, this LGTM 👍 

_Feedback using Android Simulator Pixel 2 XL running Android 10._

**Pull request merged**